File: leetcode/804.cpp
#include <iostream>
#include <string>
#include <cstring>
#include <vector>
#include <set>

using namespace std;

string moscode[] = {".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};

int fun(vector<string> &words)
{
    set<string> myset = {};
    for(int i = 0; i < words.size(); i++)
    {
        string s = words[i];//c_str();
        //cout << s << endl;
        string ss = "";
        for(int j = 0; j < s.size(); j++)
        {
            //cout << moscode[s[j]-'a'] << endl;
            ss = ss + moscode[s[j]-'a'];
        }
        myset.insert(ss);
    }
    return myset.size();
}


int main()
{
    vector<string> sss = {"gin", "zen", "gig", "msg"};
    //cout << moscode[0] << endl;
    //cout << sss[0] << endl;
    cout << fun(sss) << endl;

    return 0;
}

File: leetcode/804.py
def fun(words):
    lis1 = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]
    lis2 = [chr(a) for a in range(ord('a'),ord('a')+26,1)]
    lisz = zip(lis2,lis1)
    liDict = dict((a,b) for a,b in lisz)
    ans = []
    for word in words:
        str = ''
        for i in range(len(word)):
            str = str + liDict[word[i]]
        ans.append(str)
        #print(ans)
    return len(list(set(ans)))

wordsss = ["gin", "zen", "gig", "msg"]
print(fun(wordsss))
